drivers/soundwire/bus.c

```diff
@@ -326,13 +326,24 @@ int sdw_transfer_defer(struct sdw_bus *bus, struct sdw_msg *msg,
 int sdw_fill_msg(struct sdw_msg *msg, struct sdw_slave *slave,
 		 u32 addr, size_t count, u16 dev_num, u8 flags, u8 *buf)
 {
+	u8 addr_page1;
+	u8 addr_page2;
+
 	memset(msg, 0, sizeof(*msg));
 	msg->addr = addr; /* addr is 16 bit and truncated here */
 	msg->len = count;
 	msg->dev_num = dev_num;
 	msg->flags = flags;
 	msg->buf = buf;
 
+	if (slave) {
+		slave->rw_transactions++;
+		if (flags & SDW_MSG_FLAG_READ)
+			slave->read_transactions++;
+		if (flags & SDW_MSG_FLAG_WRITE)
+			slave->write_transactions++;
```

Comment:
As 
```
enum {
        SDW_MSG_FLAG_READ = 0,
        SDW_MSG_FLAG_WRITE,
};
```
I think it should be
```
                if (flags & SDW_MSG_FLAG_WRITE)
                        slave->write_transactions++;
                else
                        slave->read_transactions++;
```
With this change the value of rw_transactions is equal to read_transactions + write_transactions.

Reply:
@plbossart @bardliao using `flags` is actually a bit confusing for a simple `enum` value. Flags are usually bitfields, and here it is just one of a set of (presumably) mutually exclusive values. So, if it is an `enum` it rather shouldn't be called `flags`, but `mode`? `direction`? etc. And then you shouldn't use `&` here at all but just a plain simple `==`. If it should be flags though, it's use should be changed to, e.g. `.flags = BIT(SDW_MSG_FLAG_WRITE)` etc.

Reply:
Darn it, I keep making this mistake because of this stupid flags definition. we may need to change this. Thanks @bardliao for pointing this out.